src/common/vector_operations/vector_cast.cpp

```diff
@@ -12,6 +12,43 @@
 #include "duckdb/common/limits.hpp"
 namespace duckdb {
 
+static bool ToUnionCast(Vector &source, Vector &result, idx_t count, string *error_message) {
+	auto &candidates = UnionType::GetChildrenOfType(result.GetType(), source.GetType());
+
+	if(candidates.size() == 0) {
+		throw TypeMismatchException(source.GetType(), result.GetType(), "Cannot cast to UNION");
+	}
+	else if(candidates.size() == 1){
```

Comment:
Prefer early-out to make code more readable/easier to follow: `if (candidates.size() != 1) { throw ... } `